packages/webdriverio/src/commands/browser/mock.ts

```diff
@@ -100,34 +100,36 @@ import DevtoolsNetworkInterception from '../../utils/interception/devtools'
 import WebDriverNetworkInterception from '../../utils/interception/webdriver'
 import { getBrowserObject } from '../../utils'
 
-const SESSION_MOCKS: Set<Interception> = new Set()
+const SESSION_MOCKS: Map<string, Set<Interception>> = new Map()
 
-export default async function mock (this: WebdriverIO.BrowserObject, url: string, filterOptions: WebdriverIO.MockFilterOptions) {
+export default async function mock(this: WebdriverIO.BrowserObject, url: string, filterOptions: WebdriverIO.MockFilterOptions) {
     const NetworkInterception = this.isSauce ? WebDriverNetworkInterception : DevtoolsNetworkInterception
-
     if (!this.isSauce) {
         await this.getPuppeteer()
     }
-
+    const browser = getBrowserObject(this)
+    const windowHandle: string = await browser.getWindowHandle()
     /**
      * enable network Mocking if not already
      */
-    if (SESSION_MOCKS.size === 0 && !this.isSauce) {
-        const [page] = await this.puppeteer.pages()
-        const client = await page.target().createCDPSession()
-        await client.send('Fetch.enable', {
-            patterns: [{ requestStage: 'Request' }, { requestStage: 'Response' }]
+    if (!SESSION_MOCKS.has(windowHandle) && !this.isSauce) {
```

Comment:
I'd better always define a Set per window and check is it empty like https://github.com/webdriverio/webdriverio/compare/mock-multi-window#diff-03e7544b40f7605d80b9c64844538cd8a50ccfdddf92b3c452f30b5fe43bb78eR114

I think this is what @christian-bromann is asking for